src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/SetPoolForm/SetPoolFormFields/SetPoolFormFields.test.tsx

```diff
@@ -46,57 +39,31 @@ describe("SetPoolFormFields", () => {
   });
 
   it("shows a select if select pool radio chosen", async () => {
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
-        >
-          <CompatRouter>
-            <SetPoolForm
-              clearSidePanelContent={jest.fn()}
-              machines={[]}
-              processingCount={0}
-              viewingDetails={false}
-            />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
+    renderWithBrowserRouter(
+      <SetPoolForm
+        clearSidePanelContent={jest.fn()}
+        machines={[]}
+        processingCount={0}
+        viewingDetails={false}
+      />,
+      { route, state }
     );
-    await act(async () => {
-      wrapper.find("[data-testid='select-pool'] input").simulate("change", {
-        target: { name: "poolSelection", value: "select" },
-      });
-    });
-    wrapper.update();
-    expect(wrapper.find("Select").exists()).toBe(true);
+    await userEvent.click(screen.getByLabelText("Select pool"));
+    expect(screen.getByRole("combobox")).toBeInTheDocument();
```

Comment:
It would be better if you could check that the combobox is not displayed before clicking the input on the line above (and use a more specific query with a combobox label if possible).